### ISSUE TITLE
Is the decoder mask not used in loss function?

### ISSUE THREAD

**Issue by BitMindLab** [pytorch/fairseq#1431]
### some decoder examples
https://github.com/pytorch/fairseq/blob/5349052aae4ec1350822c894fbb6be350dff61a0/fairseq/models/lstm.py#L429-L430

https://github.com/pytorch/fairseq/blob/5349052aae4ec1350822c894fbb6be350dff61a0/fairseq/models/transformer.py#L687-L693

`[PAD]` may exist in target sentence. 
That is to say, `x`  contains hidden state of `[PAD]`.

### loss functioin example
https://github.com/pytorch/fairseq/blob/5349052aae4ec1350822c894fbb6be350dff61a0/fairseq/criterions/label_smoothed_cross_entropy.py#L68-L75

Why not remove `[PAD]` by decoder mask?

### my question

Is the decoder mask not used in loss function?

That is to say, `[PAD]` will affect backward process?

**Comment by myleott** [comment]
The loss function ignores padding symbols (see the `ignore_index` kwarg), so there won't be any gradients generated from these positions. We also restrict the self-attention in transformer to exclude pad symbols. Is that what you mean?

**Comment by BitMindLab** [comment]
👍  thanks for your reply.

https://github.com/pytorch/fairseq/blob/c81fed46ac7868c6d80206ff71c6f6cfe93aee22/fairseq/criterions/label_smoothed_cross_entropy.py#L18-L21
